docs/howtos/adding-a-new-component.md

```diff
@@ -0,0 +1,161 @@
+
+```
+[build-dependencies]
+uniffi_build = { version = "<latest version here>", features=["builtin-bindgen"] }
+```
+
+Include your new crate in the application-services workspace, by adding
+it to the `members` and `default-members` lists in the `Cargo.toml` at
+the root of the repository.
+
+In order to be published to consumers, your crate must be included in the
+["megazord"](../design/megazords.md) crate for each target platform:
+
+* For Android, add it as a dependency in `./megazords/full/Cargo.toml` and
+  add a `pub use <your_crate_name>` to `./megazords/full/src/lib.rs`.
+* For iOS, add it as a dependency in `./megazords/ios/rust/Cargo.toml` and
+  add a `pub use <your_crate_name>` to `./megazords/ios/rust/src/lib.rs`.
+
+Run `cargo check -p <your_crate_name>` in the repository root to confirm that
+things are configured properly. This will also have the side-effect of updating
+`Cargo.lock` to contain your new crate.
+
+
+## The Kotlin Bindings
+
+Make a `./components/<your_crate_name>/android` subdirectory to contain
+Kotlin- and Android-specific code. This directory will contain a gradle
+project for building your Kotlin bindings.
+
+Copy the `build.gradle` file from `./components/crashtest/android/` into
+your own component's directory, and edit it to replace the references to
+`crashtest.udl` with your own component's `.udl` file.
+
+Create a file `./components/<your_crate_name>/uniffi.toml` with the
+following contents:
+
+```
+[bindings.kotlin]
+package_name = "mozilla.appservices.<your_crate_name>"
+cdylib_name = "megazord"
+```
+
+Create a file `./components/<your_crate_name>/android/src/main/AndroidManifest.xml`
+with the following contents:
+
+```
+<manifest xmlns:android="http://schemas.android.com/apk/res/android"
+    package="org.mozilla.appservices.<your_crate_name>" />
+```
+
+In the root of the repository, edit `.buildconfig-android.yml`to add
+your component's metadata. This will cause it to be included in the
+gradle workspace and in our build and publish pipeline. Check whether
+it builds correctly by running:
+* `./gradlew <your_crate_name>:assembleDebug`
+
+You can include hand-written Kotlin code alongside the automatically
+generated bindings, by placing `.kt`` files in a directory named:
+* `./android/src/test/java/mozilla/appservices/<your_crate_name>/`
+
+You can write Kotlin-level tests that consume your component's API,
+by placing `.kt`` files in a directory named:
+* `./android/src/test/java/mozilla/appservices/<your_crate_name>/`.
+
+Run your component's Kotlin tests with `./gradlew <your_crate_name>:test`.
+
+
+## The Swift Bindings
+
+Make a `./components/<your_crate_name>/ios` subdirectory to contain
+Swift- and iOS-specific code. The UniFFI-generated swift bindings will
+be written to a subdirectory named `Generated`.
+
+You can include hand-written Swift code alongside the automatically
+generated bindings, by placing `.swift` files in a directory named:
+* `./ios/<your_crate_name>/`
```

Comment:
It's hard for me to create the mental model from the text above.  Maybe just show a list of the directories under iOS for an Example component with a line of text describing what each one is for, rather than writing prose?

Reply:
Thanks, good idea, I will add this for both the iOS and Android parts 👍